storage/innobase/log/log0log.cc

```diff
@@ -804,7 +808,7 @@ void log_t::resize_write_buf(size_t length) noexcept
   }
 
   ut_a(os_file_write_func(IORequestWrite, "ib_logfile101", resize_log.m_file,
-                          buf, offset, length) == DB_SUCCESS);
+                          b, offset, length) == DB_SUCCESS);
```

Comment:
In 4ca355d863e3b6a14439eebbb2958afccb3548e3 `log_t::write_buf()` was optimized so that in case the last written log block is incomplete `(length <= write_size_1)`, we will keep filling the current `log_sys.buf` or `log_sys.resize_buf`. This will avoid a `memcpy` of up to several kilobytes per call when the same block is being written over and over until it is completely filled.

I would recommend you to run the modified test `innodb.log_file_size_online` some 100 times, with some of these changes reverted. If you revert any of the changes to `innodb_log_file_size_update()`, `log_t::resize_start()`, `log_t::resize_write_buf()`, `log_t::write_buf()`, the test will start to fail.